### PR TITLE
Bump k8s dep to v1.26

### PR DESCRIPTION
**Pull request by PiotrProkop**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

bump k8s dep to v1.26.2
adapt to k/k interface changes

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by PiotrProkop** [comment]
@Huang-Wei feel free to close this PR if you feel I have jump-started this effort. I know that `v0.25.5` is not yet released but I wanted to check how much effort would it take to bump k8s to `v0.26.1`.

File: apis/config/scheme/scheme_test.go [hunk 1]
```diff
@@ -720,7 +720,6 @@ leaderElection:
   retryPeriod: 0s
 metricsBindAddress: ""
 parallelism: 0
-percentageOfNodesToScore: 0
```
Comment (PiotrProkop) [inline apis/config/scheme/scheme_test.go:723]: this field got converted to `*int64` which default value is `nil` and it uses `omitempty` tag

File: apis/config/scheme/scheme_test.go [hunk 2]
```diff
@@ -77,12 +77,12 @@ profiles:
         weight: 1
   - name: TargetLoadPacking
     args:
-      targetUtilization: 60
```
Comment (PiotrProkop) [inline apis/config/scheme/scheme_test.go:80]: my text editor removes trailing whitespaces, I can revert those changes if needed.

File: pkg/capacityscheduling/capacity_scheduling.go [hunk 3]
```diff
@@ -840,7 +838,7 @@ func computePodResourceRequest(pod *v1.Pod) *framework.Resource {
 	}
 
 	// If Overhead is being utilized, add to the total requests for the pod
-	if pod.Spec.Overhead != nil && utilfeature.DefaultFeatureGate.Enabled(kubefeatures.PodOverhead) {
```
Comment (PiotrProkop) [inline pkg/capacityscheduling/capacity_scheduling.go:843]: `PodOverhead` is GA in 1.26

**Comment by Huang-Wei** [comment]
> @Huang-Wei feel free to close this PR if you feel I have jump-started this effort. I know that v0.25.5 is not yet released but I wanted to check how much effort would it take to bump k8s to v0.26.1.

I haven't started the work as we usually give 2~3 months for one release to develop. Given 0.24 was released in mid Dec, I'd say giving it ~1 month and then we should be able to bump to 0.26. WDYT?

**Comment by PiotrProkop** [comment]
> > @Huang-Wei feel free to close this PR if you feel I have jump-started this effort. I know that v0.25.5 is not yet released but I wanted to check how much effort would it take to bump k8s to v0.26.1.
> 
> I haven't started the work as we usually give 2~3 months for one release to develop. Given 0.24 was released in mid Dec, I'd say giving it ~1 month and then we should be able to bump to 0.26. WDYT?

Sounds good! I wasn't aware of the cadence, so should I leave this PR open and we will update it in a month or should I close it and you will open new one?

For features requiring k8s 1.26, I'll just open them and note that they are blocked by this PR 😄

**Comment by Huang-Wei** [comment]
> so should I leave this PR open and we will update it in a month or should I close it and you will open new one?

We have just leave this one open :)

**Comment by PiotrProkop** [comment]
@Huang-Wei I have updated deps to k8s 0.26.2, which was released yesterday. Can I get `ok-to-test`?

**Comment by Huang-Wei** [comment]
Thanks @PiotrProkop I will start cutting the v0.25.x release early next week. After that, I will merge this PR into master.

File: pkg/coscheduling/coscheduling_test.go [hunk 4]
```diff
@@ -39,6 +39,11 @@ import (
 	testutil "sigs.k8s.io/scheduler-plugins/test/util"
 )
 
+func createPodInfo(pod *v1.Pod) *framework.PodInfo {
```
Comment (Huang-Wei) [inline pkg/coscheduling/coscheduling_test.go:42]: nit: move it to a shared place and renamed to `func mustNewPodInfo(t *testing.T, pod *v1.Pod) *framework.PodInfo {`, so we can fatalf the non-nil error.

**Review by Huang-Wei — COMMENTED** [review]
Just one nit. LGTM otherwise.

File: pkg/coscheduling/coscheduling_test.go (hunk 4 above)
Comment (PiotrProkop) [reply pkg/coscheduling/coscheduling_test.go]: done!

**Comment by Huang-Wei** [comment]
/lgtm
/approve

Thanks @PiotrProkop !

**Pull request merged**